lib/ecto/migrator.ex

```diff
@@ -42,8 +42,9 @@ defmodule Ecto.Migrator do
   def migrated_versions(repo, opts \\ []) do
     verbose_schema_migration repo, "retrieve migrated versions", fn ->
       SchemaMigration.ensure_schema_migrations_table!(repo, opts[:prefix])
-      SchemaMigration.migrated_versions(repo, opts[:prefix])
```

Comment:
You have removed the `lock_for_migrations` outside of the `verbose_schema_migration` function but I think it should still be inside. Otherwise we lose the helpful error message that is part of `verbose_schema_migration` - which is meant to add folks with a bad schema_migration setup.